app/scripts/controllers/preferences-controller.ts

```diff
@@ -1016,6 +1018,21 @@ export class PreferencesController extends BaseController<
     });
   }
 
+  getDisabledAccountUpgradeChains(): string[] {
+    return this.state.accountUpgradeDisabledChains ?? [];
+  }
+
+  disableAccountUpgradeForChain(chainId: string): void {
+    this.update((state) => {
+      const { accountUpgradeDisabledChains: existingDisabledChains } = state;
+
+      state.accountUpgradeDisabledChains = [
+        ...(existingDisabledChains ?? []),
+        chainId,
+      ];
```

Comment:
```suggestion
      if (!existingDisabledChains?.includes(chainId)) {
         state.accountUpgradeDisabledChains = [
           ...(existingDisabledChains ?? []),
           chainId,
         ];
      }
```

Would it make sense to add simple if to prevent duplicates?

Reply:
Good spot, done.